kvx/src/implementations/disk.rs

```diff
@@ -348,63 +346,33 @@ impl PathBufExt for PathBuf {
 
 #[derive(Debug)]
 struct FileLock {
-    file: File,
-    lock_path: PathBuf,
+    lock: fd_lock::RwLock<File>,
 }
 
 impl FileLock {
-    const POLL_LOCK_INTERVAL: Duration = Duration::from_millis(10);
-
-    pub fn lock(path: impl AsRef<Path>) -> Result<Self> {
+    fn create(path: impl AsRef<Path>) -> Result<Self> {
         let path = path.as_ref();
 
-        if !path.try_exists().unwrap_or_default() {
-            fs::create_dir_all(path)?;
-        }
-
         let lock_path = path.join(LOCK_FILE_NAME);
 
-        let file = loop {
-            let file = OpenOptions::new()
-                .create_new(true)
-                .read(true)
-                .write(true)
-                .open(&lock_path);
-
-            match file {
-                Ok(file) => break file,
-                _ => thread::sleep(Self::POLL_LOCK_INTERVAL),
-            };
+        let lock_file = if lock_path.exists() {
+            File::open(lock_path)?
+        } else {
+            if !path.try_exists().unwrap_or_default() {
```

Comment:
Do you need `create_dir_all` here or should all the parents exist already?

In either case, I would probably just skip the check and either unconditionally call `create_dir_all` or call `create_dir` unconditionally and ignore the `AlreadyExists` error.

Reply:
The parent dir may not exist, depending on the scope. I kept the path check, because I think we should care about a failure to create the directory path. But, if you want I can change it to accept the `AlreadyExists` error only, but error out on others - using some matching.

Reply:
`create_dir_all` doesn’t error if the path exists, so you don’t even need that and could just call it. Not sure about the overhead, though, or whether it matters.